### PR TITLE
Fix hc document issues

### PR DESCRIPTION
**Pull request by Harsimran1**
https://gesundheitscloud.atlassian.net/browse/HC-1529

* Change "identifier" to "id" in hcAuthor to sync with other SDKs
* Return "specialty" and "author" field in author on download if added during upload (currenly returns undefined)
* Return proper additionalIds field.
* Make "author" optional as in other SDKs (currently throws an exception if "identifier" is not passed.)

### CODE REVIEW

File: src/lib/models/utils/hcDocumentUtils.js [hunk 1]
```diff
@@ -69,11 +77,24 @@ const hcDocumentUtils = {
...
         fhirObject.content = hcDocument.attachments ? hcDocument.attachments.map(attachment =>
             ({ attachment: hcAttachmentUtils.toFhirObject(attachment) })) : [];
 
-        fhirObject.contained = [hcAuthorUtils.toFhirObject(hcDocument.author, clientId)];
+        if (hcDocument.author) {
+            fhirObject.contained.push(hcAuthorUtils.toFhirObject(hcDocument.author, clientId));
+        }
+
+        // Information about where the document was created.
+        if (hcDocument.author && hcDocument.author.specialty) {
```
Comment (ibihim) [inline src/lib/models/utils/hcDocumentUtils.js]: What about? Not a change request, just a question about your opinion. I am fine with this too.
```JavaScript
if (hcDocument.additionalIds) {
    fhirObject.identifier = Object.keys(hcDocument.additionalIds).map(key => ({
        value: hcDocument.additionalIds[key],
        assigner: {
            reference: key,
        },
    }));
}

if (!hcDocument.author) {
    return;
}

fhirObject.contained.push(hcAuthorUtils.toFhirObject(hcDocument.author, clientId));

// Information about where the document was created.
if (hcDocument.author.specialty) {
    fhirObject.context = {
        practiceSetting: hcSpecialtyUtils.toFhirCodeableConcept(
            hcDocument.author.specialty,
        ),
    };
}
```

File: src/lib/models/utils/hcSpecialtyUtils.js [hunk 2]
```diff
@@ -124,6 +124,25 @@ const hcSpecialtyUtils = {
...
         return Object.keys(Specialty).map(key => parseInt(key));
     },
+
+    // Specialty is not resourceType in FHIR but is often used as codeable concept
+    // We use it to define the practice setting in which document was created
+    // Speciality in our model is a single value.So the coding array always has one
+    // value.
+    fromFhirCodeableConcept(fhirCodeableConcept) {
+        return fhirCodeableConcept.coding && fhirCodeableConcept.coding[0]
+            ? fhirCodeableConcept.coding[0].code : undefined;
```
Comment (ibihim) [inline src/lib/models/utils/hcSpecialtyUtils.js]: :smile:

Comment (ibihim) [reply src/lib/models/utils/hcSpecialtyUtils.js]: `fhirCodeableConcept.coding && fhirCodeableConcept.coding[0] ? fhirCodeableConcept.coding[0].code : undefined;`
should be same as 
`fhirCodeableConcept.coding && fhirCodeableConcept.coding[0] && fhirCodeableConcept.coding[0].code`
correct me if i am wrong (headache)

File: src/lib/models/utils/hcDocumentUtils.js (hunk 1 above)
Comment (Harsimran1) [reply src/lib/models/utils/hcDocumentUtils.js]: I want to return the fhirObject. 
```JavaScript
if (!hcDocument.author) {
    return fhirObject;
}
```
should do.

**Pull request merged**